cachi2/core/package_managers/generic.py

```diff
@@ -1,13 +1,54 @@
-from cachi2.core.errors import PackageRejected
+import logging
+from typing import Literal, Optional
+
+import yaml
+from pydantic import BaseModel, ValidationError, field_validator
+
+from cachi2.core.errors import PackageManagerError, PackageRejected
 from cachi2.core.models.input import Request
 from cachi2.core.models.output import RequestOutput
 from cachi2.core.models.sbom import Component
 from cachi2.core.rooted_path import RootedPath
 
+LOG = logging.getLogger(__name__)
 DEFAULT_LOCKFILE_NAME = "cachi2_generic.yaml"
 DEFAULT_DEPS_DIR = "deps/generic"
 
 
+class LockfileMetadata(BaseModel):
```

Comment:
I wonder if it is not worth having the model definitions in a standalone file (even if they're small now). This way, we'd make this into a package, with at least two modules (one for the models, other for the resolving itself), and probably a better foundation for expanding later on.

Reply:
Sounds good to me.

Reply:
refactored into two modules